### PR TITLE
[web] Add styling in CSS reset to remove Edge password reveal icon

### PR DESCRIPTION
**Pull request by htoor3**
Microsoft Edge includes a password reveal icon pseudoelement inside text inputs by default.  To allow users to add custom icons to their password inputs and to keep behavior consistent with Chrome and Firefox, we should remove the pseudoelement in password inputs for Edge browsers.

Fixes https://github.com/flutter/flutter/issues/83695

Note for tests: `getComputedStyle()` doesn't seem to work with this pseudoelement `::-ms-reveal` in newer versions of Edge, so there doesn't appear to be a way to add a unit test to determine that this style was applied. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/embedder.dart [hunk 1]
```diff
@@ -541,6 +541,7 @@ void applyGlobalCssRulesToSheet(
 }) {
   final bool isWebKit = browserEngine == BrowserEngine.webkit;
   final bool isFirefox = browserEngine == BrowserEngine.firefox;
+  final bool isEdge = domWindow.navigator.userAgent.contains('Edg/');
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/embedder.dart:544]: Consider moving this to `browser_detection.dart`.

**Review by yjbanov — COMMENTED** [review]
LGTM if LGT @nbayati

**Comment by skia-gold** [comment]
Gold has detected about 72 new digest(s) on patchset 4.
View them at https://flutter-engine-gold.skia.org/cl/github/38011

File: lib/web_ui/lib/src/engine/host_node.dart [hunk 2]
```diff
@@ -111,6 +111,19 @@ class ShadowDomHostNode implements HostNode {
       browserEngine: browserEngine,
       hasAutofillOverlay: browserHasAutofillOverlay(),
     );
+
+    // Removes password reveal icon for text inputs in Edge browsers.
+    // Style tag w/styles needs to be injected into DOM because non-Edge
```
Comment (nbayati) [inline lib/web_ui/lib/src/engine/host_node.dart:116]: nit: "Style tag w/styles" sounds a bit ambiguous. Maybe rewording it and using the full word "with" instead of "w/" would help?

**Review by nbayati — APPROVED** [review]
Just a nit. Otherwise, LGTM.

**Comment by skia-gold** [comment]
Gold has detected about 38 new digest(s) on patchset 10.
View them at https://flutter-engine-gold.skia.org/cl/github/38011

**Comment by erojostt** [comment]
Apparently the issue is not resolved with the following versions of Flutter and Microsoft Edge:

- Flutter version 3.7.11

- Microsoft Edge Version

image


image

**Comment by lclaesen** [comment]
> Apparently the issue is not resolved with the following versions of Flutter and Microsoft Edge:
> 
> * Flutter version 3.7.11
> * Microsoft Edge Version
> 
> image
> 
> image

As per the [release notes](https://docs.flutter.dev/release/release-notes/release-notes-3.10.0), the fix for this lands in Flutter v3.10.0.
